src/hssm/hssm.py

```diff
@@ -203,15 +215,15 @@ def __init__(
         # Check if model has default config
         if _model_has_default(self.model_name, self.loglik_kind):
             model = cast(SupportedModels, model)
-            default_config = default_model_config[model][loglik_kind]
+            default_config = deepcopy(default_model_config[model][loglik_kind])
 
             self.model_config = (
                 default_config
                 if model_config is None
                 else merge_dicts(default_config, model_config)
             )
             self.loglik = self.model_config["loglik"] if loglik is None else loglik
-            self.list_params = default_params[model]
+            self.list_params = default_params[model][:]
```

Comment:
what's the effect of this change?

Reply:
A priori seems equivalent

Reply:
This is actually important. We need to create a copy, because `list_params` are lists. Without creating a copy any changes to this object will affect all models pointing to the same object. It was quite a sneaky bug to uncover.

Reply:
got it thanks